### PR TITLE
breaking(liveness): update onError callback to use LivenessError type instead of Error

### PR DESCRIPTION
**Pull request by thaddmt**
#### Description of changes
- Breaking change to `onError` prop in liveness component API
- Updates to use `LivenessError` instead of `Error` type so we can pass more information from server errors
- Updated documentation as well

#### Checklist

- [ ] Have read the [Pull Request Guidelines](https://github.com/aws-amplify/amplify-ui/blob/main/CONTRIBUTING.md)
- [ ] PR description included
- [ ] Relevant documentation is changed or added (and PR referenced)
- [ ] `yarn test` passes and tests are updated/added
- [ ] No side effects or [`sideEffects`](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react/CONTRIBUTING.md#code-standards) field updated

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Comment by reesscot** [comment]
Can you please update the migration guide for React?

File: .changeset/rude-bananas-sneeze.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@aws-amplify/ui-react-liveness": major
```
Comment (zchenwei) [inline .changeset/rude-bananas-sneeze.md:2]: Does that mean we plan to major bump the package next time?

**Review by zchenwei — DISMISSED** [review]
LGTM

File: .changeset/rude-bananas-sneeze.md (hunk 1 above)
Comment (thaddmt) [reply .changeset/rude-bananas-sneeze.md:2]: yep!

File: docs/src/pages/[platform]/connected-components/liveness/full-api.react.mdx [hunk 2]
```diff
@@ -22,7 +22,13 @@ Below is the full list of props that can be used with the `FaceLivenessDetectorC
 
 <ReactPropsTable props={FACE_LIVENESS_DETECTOR_CONFIG} />
 
-### Error States
+### LivenessError
+
+Below is the LivenessError type used in the `onError` function that should give you more granularity into the types of errors happening in the component. You can also reference the [source code here](https://github.com/aws-amplify/amplify-ui/blob/main/packages/react-liveness/src/components/FaceLivenessDetector/service/types/liveness.ts).
```
Comment (wlee221) [inline docs/src/pages/[platform]/connected-components/liveness/full-api.react.mdx:27]: Is the source code link necessary if `ReactPropsTable>` is there? Just worried we'll break the link unintentionally in the future.

Comment (thaddmt) [reply docs/src/pages/[platform]/connected-components/liveness/full-api.react.mdx:27]: Probably not! I guess we included it here to cover all bases but im happy to remove

Comment (wlee221) [reply docs/src/pages/[platform]/connected-components/liveness/full-api.react.mdx:27]: Sounds good!

**Review by wlee221 — APPROVED** [review]

**Review by hbuchel — DISMISSED** [review]
LGTM!

**Review by wlee221 — DISMISSED** [review]

**Review by hbuchel — APPROVED** [review]

File: docs/src/pages/[platform]/getting-started/migration/migration.react.mdx [hunk 3]
```diff
@@ -320,3 +321,53 @@ The latest version of the `Authenticator` has a completely different set of CSS
...
+      )}
+    </ThemeProvider>
+  );
+);
+
+```
+
+#### CDN CSP Policy
+
+The 2.x version of the `FaceLivenessDetector` has updated the default CDN for TensorFlow and Blazeface to a new URL. If your application has an existing CSP policy ensure to update the policy to allow https://cdn.liveness.rekognition.amazonaws.com. Please look over the [Liveness Config](../connected-components/liveness/customization#facelivenessdetectorconfig) section for more information.
```
Comment (reesscot) [inline docs/src/pages/[platform]/getting-started/migration/migration.react.mdx:373]:
```suggestion
The 2.x version of the `FaceLivenessDetector` has updated the default CDN for TensorFlow and Blazeface to a new URL. If your application has an existing CSP policy, update your policy to allow https://cdn.liveness.rekognition.amazonaws.com. Please look over the [Liveness Config](../connected-components/liveness/customization#facelivenessdetectorconfig) section for more information.
```

**Review by reesscot — APPROVED** [review]

**Review by zchenwei — APPROVED** [review]

**Pull request merged**